pynamodb/attribute_dict.py

```diff
@@ -0,0 +1,33 @@
+import collections
+
+
+class AttributeDict(collections.MutableMapping):
```

Comment:
i just pulled this class out, not sure how much refactoring of it i should really be doing